tools/autograd/templates/Functions.cpp

```diff
@@ -90,6 +90,13 @@ int64_t _safe_size(IntArrayRef sizes, IntArrayRef dim) {
   return size;
 }
 
+Tensor sqrt_euclidean_dist_backward(const Tensor & grad, const Tensor & self) {
+  // handle case at 0 where we return a subgradient containing 0
+  Tensor result = grad / (2 * self);
+  result.masked_fill_(self == 0, 0);
+  return result;
```

Comment:
@albanD suggested the following, which should be faster than masked_fill, but we did not benchmark. 
```
Tensor  scale_v;
// If norm is very small, self is also. So the result will be very small anyway
scale_v = grad / (2 * self.clamp_min(eps));
return self * scale_v;
```

Reply:
Note that this was for norm. Won't apply as-is here because we don't have `self` here but `x1 - x2` :/